NPE when related schema definition missing (Version: 1.8.0)

this is both a question and an issue

given this configuration
```kt
  tasks.withType<GraphQLCodegenGradleTask> {
    outputDir = File("$buildDir/generated")
  }
```

I have 2 modules in a multi-module setup, one `:util` contains a `root.graphqls` that contains basic types, in another module `:graph:iam` which depends on `implementation(project(":util"))` which has its own schema. When I ran it I got this

```
> gw graphqlCodegen --stacktrace                             # bragginrights/webservice -> feature/GH-3 ! ? RC=1
Using gradle at '/Users/calebcushing/IdeaProjects/bragginrights/webservice/gradlew' to run buildfile '/Users/calebcushing/IdeaProjects/bragginrights/webservice/build.gradle.kts':

> Task :graphql:iam:graphqlCodegen FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':graphql:iam:graphqlCodegen'.
> java.lang.NullPointerException (no error message)

* Try:
Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':graphql:iam:graphqlCodegen'.
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.lambda$executeIfValid$1(ExecuteActionsTaskExecuter.java:207)
        at org.gradle.internal.Try$Failure.ifSuccessfulOrElse(Try.java:263)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeIfValid(ExecuteActionsTaskExecuter.java:205)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:186)
        at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:114)
        at org.gradle.api.internal.tasks.execution.FinalizePropertiesTaskExecuter.execute(FinalizePropertiesTaskExecuter.java:46)
        at org.gradle.api.internal.tasks.execution.ResolveTaskExecutionModeExecuter.execute(ResolveTaskExecutionModeExecuter.java:62)
        at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:57)
        at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:56)
        at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:36)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:77)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:55)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:52)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:416)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:406)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:165)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:250)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:158)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:102)
        at org.gradle.internal.operations.DelegatingBuildOperationExecutor.call(DelegatingBuildOperationExecutor.java:36)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:52)
        at org.gradle.execution.plan.LocalTaskNodeExecutor.execute(LocalTaskNodeExecutor.java:41)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:370)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:357)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:350)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:336)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.lambda$run$0(DefaultPlanExecutor.java:127)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.execute(DefaultPlanExecutor.java:191)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.executeNextNode(DefaultPlanExecutor.java:182)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.run(DefaultPlanExecutor.java:124)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
Caused by: java.lang.NullPointerException
        at com.kobylynskyi.graphql.codegen.model.definitions.ExtendedObjectTypeDefinition.getFieldDefinitions(ExtendedObjectTypeDefinition.java:13)
        at com.kobylynskyi.graphql.codegen.GraphQLCodegen.generateOperation(GraphQLCodegen.java:157)
        at com.kobylynskyi.graphql.codegen.GraphQLCodegen.processDefinitions(GraphQLCodegen.java:122)
        at com.kobylynskyi.graphql.codegen.GraphQLCodegen.generate(GraphQLCodegen.java:102)
        at io.github.kobylynskyi.graphql.codegen.gradle.GraphQLCodegenGradleTask.generate(GraphQLCodegenGradleTask.java:94)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:104)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.doExecute(StandardTaskAction.java:49)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:42)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:28)
        at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:722)
        at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:689)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$3.run(ExecuteActionsTaskExecuter.java:569)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:402)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:394)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:165)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:250)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:158)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:92)
        at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:554)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:537)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.access$300(ExecuteActionsTaskExecuter.java:108)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$TaskExecution.executeWithPreviousOutputFiles(ExecuteActionsTaskExecuter.java:278)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$TaskExecution.execute(ExecuteActionsTaskExecuter.java:267)
        at org.gradle.internal.execution.steps.ExecuteStep.lambda$execute$1(ExecuteStep.java:33)
        at org.gradle.internal.execution.steps.ExecuteStep.execute(ExecuteStep.java:33)
        at org.gradle.internal.execution.steps.ExecuteStep.execute(ExecuteStep.java:26)
        at org.gradle.internal.execution.steps.CleanupOutputsStep.execute(CleanupOutputsStep.java:67)
        at org.gradle.internal.execution.steps.CleanupOutputsStep.execute(CleanupOutputsStep.java:36)
        at org.gradle.internal.execution.steps.ResolveInputChangesStep.execute(ResolveInputChangesStep.java:49)
        at org.gradle.internal.execution.steps.ResolveInputChangesStep.execute(ResolveInputChangesStep.java:34)
        at org.gradle.internal.execution.steps.CancelExecutionStep.execute(CancelExecutionStep.java:43)
        at org.gradle.internal.execution.steps.TimeoutStep.executeWithoutTimeout(TimeoutStep.java:73)
        at org.gradle.internal.execution.steps.TimeoutStep.execute(TimeoutStep.java:54)
        at org.gradle.internal.execution.steps.CatchExceptionStep.execute(CatchExceptionStep.java:34)
        at org.gradle.internal.execution.steps.CreateOutputsStep.execute(CreateOutputsStep.java:44)
        at org.gradle.internal.execution.steps.SnapshotOutputsStep.execute(SnapshotOutputsStep.java:54)
        at org.gradle.internal.execution.steps.SnapshotOutputsStep.execute(SnapshotOutputsStep.java:38)
        at org.gradle.internal.execution.steps.BroadcastChangingOutputsStep.execute(BroadcastChangingOutputsStep.java:49)
        at org.gradle.internal.execution.steps.CacheStep.executeWithoutCache(CacheStep.java:159)
        at org.gradle.internal.execution.steps.CacheStep.execute(CacheStep.java:72)
        at org.gradle.internal.execution.steps.CacheStep.execute(CacheStep.java:43)
        at org.gradle.internal.execution.steps.StoreExecutionStateStep.execute(StoreExecutionStateStep.java:44)
        at org.gradle.internal.execution.steps.StoreExecutionStateStep.execute(StoreExecutionStateStep.java:33)
        at org.gradle.internal.execution.steps.RecordOutputsStep.execute(RecordOutputsStep.java:38)
        at org.gradle.internal.execution.steps.RecordOutputsStep.execute(RecordOutputsStep.java:24)
        at org.gradle.internal.execution.steps.SkipUpToDateStep.executeBecause(SkipUpToDateStep.java:92)
        at org.gradle.internal.execution.steps.SkipUpToDateStep.lambda$execute$0(SkipUpToDateStep.java:85)
        at org.gradle.internal.execution.steps.SkipUpToDateStep.execute(SkipUpToDateStep.java:55)
        at org.gradle.internal.execution.steps.SkipUpToDateStep.execute(SkipUpToDateStep.java:39)
        at org.gradle.internal.execution.steps.ResolveChangesStep.execute(ResolveChangesStep.java:76)
        at org.gradle.internal.execution.steps.ResolveChangesStep.execute(ResolveChangesStep.java:37)
        at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsFinishedStep.execute(MarkSnapshottingInputsFinishedStep.java:36)
        at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsFinishedStep.execute(MarkSnapshottingInputsFinishedStep.java:26)
        at org.gradle.internal.execution.steps.ResolveCachingStateStep.execute(ResolveCachingStateStep.java:94)
        at org.gradle.internal.execution.steps.ResolveCachingStateStep.execute(ResolveCachingStateStep.java:49)
        at org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep.execute(CaptureStateBeforeExecutionStep.java:79)
        at org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep.execute(CaptureStateBeforeExecutionStep.java:53)
        at org.gradle.internal.execution.steps.ValidateStep.execute(ValidateStep.java:74)
        at org.gradle.internal.execution.steps.SkipEmptyWorkStep.lambda$execute$2(SkipEmptyWorkStep.java:78)
        at org.gradle.internal.execution.steps.SkipEmptyWorkStep.execute(SkipEmptyWorkStep.java:78)
        at org.gradle.internal.execution.steps.SkipEmptyWorkStep.execute(SkipEmptyWorkStep.java:34)
        at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsStartedStep.execute(MarkSnapshottingInputsStartedStep.java:39)
        at org.gradle.internal.execution.steps.LoadExecutionStateStep.execute(LoadExecutionStateStep.java:40)
        at org.gradle.internal.execution.steps.LoadExecutionStateStep.execute(LoadExecutionStateStep.java:28)
        at org.gradle.internal.execution.impl.DefaultWorkExecutor.execute(DefaultWorkExecutor.java:33)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeIfValid(ExecuteActionsTaskExecuter.java:194)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:186)
        at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:114)
        at org.gradle.api.internal.tasks.execution.FinalizePropertiesTaskExecuter.execute(FinalizePropertiesTaskExecuter.java:46)
        at org.gradle.api.internal.tasks.execution.ResolveTaskExecutionModeExecuter.execute(ResolveTaskExecutionModeExecuter.java:62)
        at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:57)
        at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:56)
        at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:36)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:77)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:55)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:52)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:416)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:406)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:165)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:250)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:158)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:102)
        at org.gradle.internal.operations.DelegatingBuildOperationExecutor.call(DelegatingBuildOperationExecutor.java:36)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:52)
        at org.gradle.execution.plan.LocalTaskNodeExecutor.execute(LocalTaskNodeExecutor.java:41)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:370)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:357)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:350)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:336)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.lambda$run$0(DefaultPlanExecutor.java:127)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.execute(DefaultPlanExecutor.java:191)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.executeNextNode(DefaultPlanExecutor.java:182)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.run(DefaultPlanExecutor.java:124)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)


* Get more help at https://help.gradle.org

BUILD FAILED in 3s
1 actionable task: 1 executed```

moving the file to the same resources directory as `:graphql:iam` fixed the issue.

do you have any recommendations for how to do a multimodule build? or is the best answer to generate all this in 1 module.

--- a/src/main/java/com/kobylynskyi/graphql/codegen/GraphQLCodegen.java
+++ b/src/main/java/com/kobylynskyi/graphql/codegen/GraphQLCodegen.java
@@ -217,7 +217,9 @@ public class GraphQLCodegen {
 
     private void initCustomTypeMappings(Collection<ExtendedScalarTypeDefinition> scalarTypeDefinitions) {
         for (ExtendedScalarTypeDefinition definition : scalarTypeDefinitions) {
-            mappingConfig.putCustomTypeMappingIfAbsent(definition.getDefinition().getName(), "String");
+            if (definition.getDefinition() != null) {
+                mappingConfig.putCustomTypeMappingIfAbsent(definition.getDefinition().getName(), "String");
+            }
             for (ScalarTypeExtensionDefinition extension : definition.getExtensions()) {
                 mappingConfig.putCustomTypeMappingIfAbsent(extension.getName(), "String");
             }

--- a/src/main/java/com/kobylynskyi/graphql/codegen/model/definitions/ExtendedDefinition.java
+++ b/src/main/java/com/kobylynskyi/graphql/codegen/model/definitions/ExtendedDefinition.java
@@ -1,12 +1,15 @@
 package com.kobylynskyi.graphql.codegen.model.definitions;
 
-import graphql.language.*;
+import graphql.language.Comment;
+import graphql.language.NamedNode;
+import graphql.language.Node;
 import lombok.Getter;
-import lombok.NonNull;
 import lombok.Setter;
 
-import java.util.*;
-import java.util.function.Function;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Objects;
 
 /**
  * Base class for all GraphQL definition types that contains base definition and its extensions
@@ -18,17 +21,23 @@ import java.util.function.Function;
 @Setter
 public abstract class ExtendedDefinition<T extends NamedNode<T>, E extends T> {
 
-    @NonNull
+    /**
+     * Nullable because some schemas can have just "extends"
+     */
     protected T definition;
     protected List<E> extensions = new ArrayList<>();
 
     public String getName() {
-        return definition.getName();
+        if (definition != null) {
+            return definition.getName();
+        } else {
+            return extensions.stream().map(NamedNode::getName).findFirst().orElse(null);
+        }
     }
 
     public List<String> getJavaDoc() {
         List<String> comments = new ArrayList<>();
-        if (definition.getComments() != null) {
+        if (definition != null && definition.getComments() != null) {
             definition.getComments().stream()
                     .map(Comment::getContent)
                     .filter(Objects::nonNull)

--- a/src/main/java/com/kobylynskyi/graphql/codegen/model/definitions/ExtendedEnumTypeDefinition.java
+++ b/src/main/java/com/kobylynskyi/graphql/codegen/model/definitions/ExtendedEnumTypeDefinition.java
@@ -4,6 +4,7 @@ import graphql.language.EnumTypeDefinition;
 import graphql.language.EnumTypeExtensionDefinition;
 import graphql.language.EnumValueDefinition;
 
+import java.util.ArrayList;
 import java.util.List;
 
 public class ExtendedEnumTypeDefinition extends ExtendedDefinition<EnumTypeDefinition, EnumTypeExtensionDefinition> {
@@ -14,7 +15,10 @@ public class ExtendedEnumTypeDefinition extends ExtendedDefinition<EnumTypeDefin
      * @return list of all enum value definitions
      */
     public List<EnumValueDefinition> getValueDefinitions() {
-        List<EnumValueDefinition> definitions = definition.getEnumValueDefinitions();
+        List<EnumValueDefinition> definitions = new ArrayList<>();
+        if (definition != null) {
+            definitions.addAll(definition.getEnumValueDefinitions());
+        }
         extensions.stream()
                 .map(EnumTypeExtensionDefinition::getEnumValueDefinitions)
                 .forEach(definitions::addAll);

--- a/src/main/java/com/kobylynskyi/graphql/codegen/model/definitions/ExtendedInputObjectTypeDefinition.java
+++ b/src/main/java/com/kobylynskyi/graphql/codegen/model/definitions/ExtendedInputObjectTypeDefinition.java
@@ -4,12 +4,16 @@ import graphql.language.InputObjectTypeDefinition;
 import graphql.language.InputObjectTypeExtensionDefinition;
 import graphql.language.InputValueDefinition;
 
+import java.util.ArrayList;
 import java.util.List;
 
 public class ExtendedInputObjectTypeDefinition extends ExtendedDefinition<InputObjectTypeDefinition, InputObjectTypeExtensionDefinition> {
 
     public List<InputValueDefinition> getValueDefinitions() {
-        List<InputValueDefinition> definitions = definition.getInputValueDefinitions();
+        List<InputValueDefinition> definitions = new ArrayList<>();
+        if (definition != null) {
+            definitions.addAll(definition.getInputValueDefinitions());
+        }
         extensions.stream()
                 .map(InputObjectTypeDefinition::getInputValueDefinitions)
                 .forEach(definitions::addAll);

--- a/src/main/java/com/kobylynskyi/graphql/codegen/model/definitions/ExtendedInterfaceTypeDefinition.java
+++ b/src/main/java/com/kobylynskyi/graphql/codegen/model/definitions/ExtendedInterfaceTypeDefinition.java
@@ -11,9 +11,11 @@ public class ExtendedInterfaceTypeDefinition extends ExtendedDefinition<Interfac
 
     public List<ExtendedFieldDefinition> getFieldDefinitions() {
         List<ExtendedFieldDefinition> definitions = new ArrayList<>();
-        definition.getFieldDefinitions().stream()
-                .map(f -> new ExtendedFieldDefinition(f, false))
-                .forEach(definitions::add);
+        if (definition != null) {
+            definition.getFieldDefinitions().stream()
+                    .map(f -> new ExtendedFieldDefinition(f, false))
+                    .forEach(definitions::add);
+        }
         extensions.stream()
                 .map(InterfaceTypeExtensionDefinition::getFieldDefinitions)
                 .flatMap(Collection::stream)

--- a/src/main/java/com/kobylynskyi/graphql/codegen/model/definitions/ExtendedObjectTypeDefinition.java
+++ b/src/main/java/com/kobylynskyi/graphql/codegen/model/definitions/ExtendedObjectTypeDefinition.java
@@ -1,6 +1,8 @@
 package com.kobylynskyi.graphql.codegen.model.definitions;
 
-import graphql.language.*;
+import graphql.language.ObjectTypeDefinition;
+import graphql.language.ObjectTypeExtensionDefinition;
+import graphql.language.Type;
 
 import java.util.ArrayList;
 import java.util.Collection;
@@ -10,9 +12,11 @@ public class ExtendedObjectTypeDefinition extends ExtendedDefinition<ObjectTypeD
 
     public List<ExtendedFieldDefinition> getFieldDefinitions() {
         List<ExtendedFieldDefinition> definitions = new ArrayList<>();
-        definition.getFieldDefinitions().stream()
-                .map(f -> new ExtendedFieldDefinition(f, false))
-                .forEach(definitions::add);
+        if (definition != null) {
+            definition.getFieldDefinitions().stream()
+                    .map(f -> new ExtendedFieldDefinition(f, false))
+                    .forEach(definitions::add);
+        }
         extensions.stream()
                 .map(ObjectTypeExtensionDefinition::getFieldDefinitions)
                 .flatMap(Collection::stream)
@@ -23,7 +27,10 @@ public class ExtendedObjectTypeDefinition extends ExtendedDefinition<ObjectTypeD
 
     @SuppressWarnings("rawtypes")
     public List<Type> getImplements() {
-        List<Type> definitionImplements = definition.getImplements();
+        List<Type> definitionImplements = new ArrayList<>();
+        if (definition != null) {
+            definitionImplements.addAll(definition.getImplements());
+        }
         extensions.stream()
                 .map(ObjectTypeDefinition::getImplements)
                 .forEach(definitionImplements::addAll);

--- a/src/main/java/com/kobylynskyi/graphql/codegen/model/definitions/ExtendedUnionTypeDefinition.java
+++ b/src/main/java/com/kobylynskyi/graphql/codegen/model/definitions/ExtendedUnionTypeDefinition.java
@@ -27,11 +27,13 @@ public class ExtendedUnionTypeDefinition extends ExtendedDefinition<UnionTypeDef
 
     private Set<String> getMemberTypeNames() {
         Set<String> memberTypeNames = new HashSet<>();
-        getDefinition().getMemberTypes().stream()
-                .map(NamedNode.class::cast)
-                .map(NamedNode::getName)
-                .forEach(memberTypeNames::add);
-        getExtensions().stream()
+        if (definition != null) {
+            definition.getMemberTypes().stream()
+                    .map(NamedNode.class::cast)
+                    .map(NamedNode::getName)
+                    .forEach(memberTypeNames::add);
+        }
+        extensions.stream()
                 .map(UnionTypeDefinition::getMemberTypes)
                 .flatMap(Collection::stream)
                 .map(NamedNode.class::cast)

--- a/src/test/java/com/kobylynskyi/graphql/codegen/GraphQLCodegenExtendTest.java
+++ b/src/test/java/com/kobylynskyi/graphql/codegen/GraphQLCodegenExtendTest.java
@@ -57,6 +57,17 @@ class GraphQLCodegenExtendTest {
     }
 
     @Test
+    void generate_onlyExtend() throws Exception {
+        schemaFinder.setIncludePattern("only-extend-queries.*\\.graphqls");
+        new GraphQLCodegen(schemaFinder.findSchemas(), outputBuildDir, mappingConfig).generate();
+
+        assertEquals(new HashSet<>(asList("Subscription.java", "UserQuery.java", "User.java",
+                "UsersCreatedSubscription.java", "CreateUserMutation.java", "Mutation.java", "Query.java",
+                "UserInput.java")), Arrays.stream(Objects.requireNonNull(outputJavaClassesDir.listFiles()))
+                .map(File::getName).collect(toSet()));
+    }
+
+    @Test
     void generateServerSideClasses_ExtensionFieldsResolvers() throws Exception {
         mappingConfig.setGenerateExtensionFieldsResolvers(true);
         new GraphQLCodegen(schemaFinder.findSchemas(), outputBuildDir, mappingConfig).generate();

--- a/src/test/resources/schemas/only-extend-queries.graphqls
+++ b/src/test/resources/schemas/only-extend-queries.graphqls
@@ -1,0 +1,20 @@
+extend type Query {
+    user(userId: ID!): User
+}
+
+extend type Mutation {
+    createUser(input: UserInput!): User
+}
+
+extend type Subscription {
+    usersCreated: [User]
+}
+
+type User {
+    id: ID!
+    name: String!
+}
+
+input UserInput {
+    name: String!
+}

